WordPress/gutenberg
# Data views table row: Make checkboxes and actions visible on touch devices

## What

Make checkboxes and quick-actions visible on touch devices.

## Why

Currently these affordances appear only when focus is within the table row, or when the table row itself is hovered. Neither of these events trigger frequently enough (or at all) on touch devices making it less obvious that users are able to perform these actions.

## How

Use media query `@media (hover: none) {` to target devices that do not support hover and override opacity styles for checkboxes and quick-actions.

## To test

On a touch device:

* Navigate to a data view 
* Switch to table layout
* Observe that checkboxes and quick-actions are visible

You can use [WordPress playground](https://playground.wordpress.net/gutenberg.html) to test this PR on a phone / tablet, or just use device previews in browser dev tools.

On other devices:

* Navigate to a data view 
* Switch to table layout
* Observe that checkboxes and quick-actions only appear when focus is within the table row, or the table row is hovered

### Before
<img width="1152" alt="Screenshot 2024-04-17 at 16 59 33" src="https://github.com/WordPress/gutenberg/assets/846565/67ed82a1-739c-4139-8946-809aefb149e4">


### After
<img width="1151" alt="Screenshot 2024-04-17 at 17 01 13" src="https://github.com/WordPress/gutenberg/assets/846565/80d5769f-1221-4916-9aae-ed82e9757fe0">
